<a href="https://colab.research.google.com/github/AlexeyKatko/ColabKatko/blob/main/%D1%82%D0%B5%D0%BE%D1%80%D0%B8%D1%8F_%D0%B2%D0%B5%D1%80%D0%BE%D1%8F%D1%82%D0%BD%D0%BE%D1%81%D1%82%D0%B8_%D0%B4%D0%BE%D0%BC%D0%B0%D1%88%D0%BD%D0%B5%D0%B5_%D0%B7%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Урок 1. Расчет вероятности случайных событий

## Задача 1


Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

### Решение 1.1
a) В колоде из 52 листов $\\С\frac{4}{52}=13$ карт одной масти.
Воспользуемся формулой подсчета количества сочетаний по $k$ элементов из множества $n$:
$\\ C^k_n=С\frac{n!}{k!(n-k)!}$

И классической формулой вероятности:
  $\\ P(A)=С\frac{m}{n}$
  
  где $m$-число **благоприятных** исходов, $n$-общее число **всех** исходов.
в нашем случае $\\P (A)=C\frac{4}{52}$-т.е. количество способов извлечь 4 карты из 52, а $\\ m=C\frac{4}{13}$, способов извлечь 4 карты одной масти.
Для облегчения расчетов напишем функцию:

In [1]:
from math import factorial
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [2]:
# Вычислим m
m=combinations(13, 4)
print(f'm = {m}')

m = 715


In [3]:
# Вычислим n
n=combinations(52, 4)
print(f'n = {n}')

n = 270725


In [4]:
# Найдем вероятность Р(4крести)
P=m/n
print(f'P(4 крести) = {round(P,4)}')

P(4 крести) = 0.0026


Либо, из классической формулы вероятности следует, что вероятность извлечь

первой картой из 4-х крестовую: 13/52,

вторую крестовую: 12/51;

третью крестовую: 11/52;

четвертую: 10/49.

А поскольку, по условию и первая, и вторая, и третья, и четвертая должны быть крестовые, то P(4крести)
Вычислим:

In [5]:
P1=13/52*12/51*11/50*10/49
print(f'P(4 крести) = {round(P1,4)}')

P(4 трефы) = 0.0026


#### Решение 1.2

б) в этом случае так же общее число исходов n=C\frac{4}{52}
, а
 будет складываться из 4 вариантов:

1. из 4-х карт 1 туз:
число способов достать 1 туз из 4-х: С^1_4
, число способов достать три остальные карты С
, общее количество сочетаний 1 туз и 3 не туза
С
из 4-х карт 2 туза:
число способов достать 2 туза из 4-х
, число способов достать две остальные карты С
, общее количество сочетаний 2 туза и 2 не туза
С
из 4-х карт 3 туза:
число способов достать 3 туза из 4-х
, число способов достать оставшуюся карту С
, общее количество сочетаний 3 туза и 1 не туз
С
из 4-х карт 4 туза:
число способов достать 4 туза из 4-х
Общее число благоприятных исходов-сумма этих 4 вариантов.
Вычислим:

In [6]:
# Общее число благоприятных исходов-сумма этих 4 вариантов.
n=combinations(52, 4)
print(f'n = {n}')

n = 270725


In [7]:
m=sum([combinations(4,1)*combinations(48,3),combinations(4,2)*combinations(48,2),combinations(4,3)*combinations(48,1),1])
print(f'm = {m}')

m = 76145


In [8]:
# Найдем вероятность Р(хотя бы 1 туз)
P=m/n
print(f'P(хотя бы 1 туз) = {round(P,4)}')

P(хотя бы 1 туз) = 0.2813


Ответ:

а) Р(4 трефы) ≈ 0.0026 ≈ 0.26%;

б) P(хотя бы 1 туз) ≈ 0.2813 ≈ 28.13%
.

# Задача 2

На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

## Решение 2.1

Так как кнопки нажимаются одновременно, то порядок не важен, и для формулы вероятности $\\ P(A)=С\frac{m}{n}$

 общее число исходов будет определяться количиством сочетаний 3-х элементов из 10, $\\ n=С\frac{3}{10}$
, а количество благоприятных исходов m=1
-замок открылся. Вычислим P(A)
:

In [9]:
P=1/combinations(10,3)
print(f'P(открыть с первой попытки) = {round(P,4)}')

P(открыть с первой попытки) = 0.0083


Ответ:

P(открыть с первой попытки) ≈ 0.0083 ≈ 0.83%

# Задача 3

В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

## Решение 3.1

Общее количество исходов будет определяться количеством сочетаний 3 деталей из 15 $\\ n=С\frac{3}{15}$
, а количество благоприятных исходов - количеством сочетаний 3-х окрашенных деталей из 9 $\\ n=С\frac{3}{9}$

In [10]:
# Р(3 из 3 окрашены)
P=combinations(9,3)/combinations(15,3)
print(f'P(3 из 3-х окрашены) = {round(P,4)}')

P(3 из 3-х окрашены) = 0.1846


ИЛИ:

In [11]:
print(f'P(3 из 3-х окрашены) = {round(9/15*8/14*7/13,4)}')

P(3 из 3-х окрашены) = 0.1846


Ответ:

P(3 из 3 окрашены) ≈ 0.1846 ≈ 18.46%

# Задача 4

В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

## Решение 4.1


  $\\ P(оба\ выигрышных)=\frac{1}{C^1_{100}}=\frac{1}{100!}/{2!(100-2)}= \frac{2!*98!}{100!}$


или

$\\P({оба\ выигрышных})=\frac{1}{100}⋅\frac{1}{99}$

In [15]:
P=1/combinations(100,2)
print(f'P(оба выигрышных = {round(P,4)})')

P(оба выигрышных = 0.0002)


Ответ:

P(оба выигрышных) ≈ 0.0002 ≈ 0.02%